### DISCUSSION TITLE
Wireguard connection status with external LED on RPi GPIO

### DISCUSSION THREAD

**Discussion by TBR-BRD** [RaspAP/raspap-webgui#1504] [Ideas]
I would like to connect an LED to a GPIO port. This LED should switch on when the Wireguard connection is successfully established. Has anyone already built and programmed this?

**Comment by billz** [comment]
You could borrow some code from [stats.py](https://github.com/RaspAP/raspap-webgui/blob/master/app/pitft/stats.py) to execute a shell command to get the WireGuard status, and combine it with code from [this PR](https://github.com/RaspAP/raspap-webgui/pull/172/files#diff-3e5f0dfc2e885a8fd7559e561462eaabf05554e972d678b1a0aa26bd779d57c8) to handle the GPIO + LED portion. Both are written in python, so a bit of cut-and-paste and some experimentation will get you there.

**Reply by coolstuff99** [reply]
Can somehow this apply as well to start or stop WG using external button attached to GPIO port?

**Reply by billz** [reply]
Sounds very similar to https://github.com/RaspAP/raspap-webgui/pull/172
Modify the actions in `button.py` to something other than `installers/reset.sh`
